testcontainers/src/core/copy.rs

```diff
@@ -0,0 +1,11 @@
+#[derive(Debug, Clone)]
+pub struct CopyToContainer {
+    pub target: String,
+    pub source: CopyDataSource,
+}
+
+#[derive(Debug, Clone)]
+pub enum CopyDataSource {
+    File(String),
+    Data(Vec<u8>),
+}
```

Comment:
This one is nice, what I could suggest:

- we can use `Path` for source, because we going to copy file from the host/caller system
- we can implement `From<Path>` (file) and `From<T: Into<Vec<u8>>` (data)
- use `impl Into<CopyDataSource>` for arguments in `ImageExt`

It will provide more flexible way to interact with API
WDYT?

Reply:
Very good idea. I imlemented a few `From<...>` but left await `From<T: Into<Vec<u8>>` because it didn't work right away.